Content.Server/GameObjects/EntitySystems/ConstructionSystem.cs

```diff
@@ -151,17 +327,241 @@ private void TryStartItemConstruction(IEntity placingEnt, string prototypeName)
             if (prototype.Stages.Count == 2)
             {
                 // Exactly 2 stages, so don't make an intermediate frame.
-                var ent = _serverEntityManager.SpawnEntity(prototype.Result, placingEnt.Transform.GridPosition);
+                var ent = EntityManager.SpawnEntity(prototype.Result, placingEnt.Transform.GridPosition);
                 hands.PutInHandOrDrop(ent.GetComponent<ItemComponent>());
             }
             else
             {
-                //TODO: Make these viable as an item and try putting them in the players hands
-                var frame = _serverEntityManager.SpawnEntity("structureconstructionframe", placingEnt.Transform.GridPosition);
+                var frame = EntityManager.SpawnEntity("structureconstructionframe", placingEnt.Transform.GridPosition);
                 var construction = frame.GetComponent<ConstructionComponent>();
-                construction.Init(prototype);
+                SetupComponent(construction, prototype);
+
+                var finalPrototype = _prototypeManager.Index<EntityPrototype>(prototype.Result);
+                if (finalPrototype.Components.TryGetValue("Item", out var itemProtoComp))
+                {
+                    if (!frame.TryGetComponent<ItemComponent>(out var itemComp))
+                        itemComp = frame.AddComponent<ItemComponent>();
+
+                    var serializer = YamlObjectSerializer.NewReader(itemProtoComp);
+                    itemComp.ExposeData(serializer);
```

Comment:
Is this not a case where you would remove the `ItemComponent` and slot in a new one?

Reply:
Basically this block of code says "If the final product has an ItemComponent, add a new ItemComponent to the intermediate frame and then load all the data from the final ItemComponent into the intermediate ItemComponent." It is loading a component up from a *different* prototype. There is no support for doing this this in the engine, and this code is pretty hacky. I'm not sure what you mean by "slot in" a new ItemComponent.

Reply:
I meant remove the old `ItemComponent` and add the new one, rather than reading data from one into the other in this way.